internal/pkg/selfupdate/update.go

```diff
@@ -22,6 +22,14 @@ func UpgradeInPlace(ctx context.Context, baseSiteURL string, channelName *string
 		}
 	}
 
+	currentBinaryPath, err := os.Executable()
+	if err != nil {
+		return fmt.Errorf("getting current binary path: %v", err)
+	}
+	if err := os.Rename(currentBinaryPath, currentBinaryPath+".old"); err != nil {
+		return fmt.Errorf("renaming binary with .old postfix: %v", err)
+	}
```

Comment:
looks like this is the same as https://github.com/humanlogio/humanlog/blob/c845ece1e849bccd7b969bd9428f718933a9bb4d/internal/pkg/selfupdate/update.go#L100-L123